### ISSUE TITLE
theme-core@0.1.5 on myst-theme integration partly breaks DataFrame rendering for React Node

### ISSUE THREAD

**Issue by GYHHAHA** [executablebooks/thebe#613]
### Describe the bug

This Bug 🐛 spent me a lot of time to find the reason. It's super wired when I saw the DataFrame is rendered twice:
<img src="https://user-images.githubusercontent.com/41546976/231040064-bf48dbd9-28f7-49e2-b6aa-242e67ae7505.png"  width="40%">
I thought it was caused by some regression from mystjs or myst-theme. However it's not. I finally got the idea from the common wrapper of these two block `<div class="thebe-output" style="position: relative;"> ... </div>` and then released maybe it was caused by some components from thebe. After I changed theme-core to 0.1.4, it works!
<img src="https://user-images.githubusercontent.com/41546976/231040646-de556266-9e29-4801-9d2b-3442f3dc5284.png"  width="40%">
Therefore this is caused by some unintended corner cases in #611 for text/html and text/plain render, which related to the fact that pandas DataFrame in the Jupyter format will both keep the original text/plain and the text/html for rendering. Hope these information will make the fix easier! @st

### Reproduce the bug

/

**Comment by rowanc1** [comment]
Thanks for the report @GYHHAHA! @stevejpurves is out on vacation this week, but when he gets back thebe and theme integration are top priorities, so we will certainly address this! Thanks also for your other PRs on the theme -- I will hopefully get to them this week!

**Comment by GYHHAHA** [comment]
Thanks for quick rely! @rowanc1 Besides, there is a dependency issue in myst-tools-theme. You can also take a look if available. Thanks!

**Comment by rowanc1** [comment]
I will keep an eye out for it (I don't think there is a specific issue tracking it?). I haven't found a good way to ensure that all of the packages work together reliably yet, and often run into installation problems when upgrading core packages. If you have suggestions or other specific issues you are running into there, happy to hear them to get a broader picture!

**Comment by GYHHAHA** [comment]
Yes, that's true. There are too many packages and they are distributed in several repos at the same time. It takes me a while to clarify the functions and relationship. Although the process was painful, fortunately I have figured out a lot. If there are some demands to document these js packages in the future, I'm willing to do the contribution.

**Comment by stevejpurves** [comment]
@GYHHAHA Thanks for the report! and I'll specifically check this case next - there were some changes to the PassiveCellRenderer to accommodate work on `ipywidgets` and ensuring smooth transition between stored and re-rendered outputs and it looks like this was introduced there.
